File: hw2/codes/cakin_haci.php
<?php
// 1. Iteration statements provided
// Iterations are provided in php by using while, for, forEach and do while loops
$bool1 = true;
$num1 = 0;
printf("1. Iteration statements provided");
printf("while -----\n");
while($bool1){
    if($num1 == 5){
        printf("num1 reached to 5\n");
        break;
    }
    printf("num1 = %d \n", $num1);
    $num1++;
}
printf("do-while -----\n");
$num2 = 0;
do{
    $num2+=2;
    printf("num2 = %d \n", $num2);
} while( $num2 < 10 );

printf("for -----\n");
for($k = 0; $k < 6; $k++){
    if($k ==4){
        printf("Reached to 4\n");
        break;
    }
    printf("k = %d \n", $k);
}
printf("for-each -----\n");
$arrayNum = array(0, 1, 2);
foreach ($arrayNum as $value) {
   $value = $value + 4;
   printf("value = %d \n", $value);
   printf( "\n");
}


// 2. Data structures suitable for iteration
// Built in data structures is in php array
printf("2. Data structures suitable for iteration\n");
$arrayNum = array(0, 1, 2, 3, 4, 5);
foreach ($arrayNum as $value) {
    printf("value = %d \n", $value);
 }
printf( "\n");
 
$birthDay = array ("day"  => 24, "month" => 02, "year"   => 2000 );
 foreach ($birthDay as $name => $value) {
    printf("value = %d \n", $value);
 }

// 3. The way the next item is accessed
printf("3. The way the next item is accessed\n");
printf( "while-----\n");
$num3 = 0;
     while( $num3 < 10 ) {
     if ( $num3 == 4){
            printf("num3 reached 4\n");
           break;
         }
     printf("num3 = %d \n", $num3);
     $num3 = $num3 + 2;

     }
     // For Loop
     printf( "\nfor-----\n");
     for( $m= 0; $m< 10 ; $m++){
        if( $m == 4 ){
            printf("m reached 4\n");
            break;
        }
        printf("m = %d \n", $m);
        $m = $m + 1;
    }

?>

File: hw2/codes/cakin_haci.go


package main

import (
	"fmt"
)

func main() {
// 1. Iteration statements provided
// Iterations are provided in go by using while, for and forEach loops
fmt.Println("1. Iteration statements provided \n")
fmt.Println("while loop \n")
var bool1 = true
trace := 0
for bool1 {
	if trace == 5 {
		fmt.Println("Trace reached to 5 \n")
		break
	}
	fmt.Println("Current value in while loop is ", trace)
	trace = trace + 1
}


fmt.Println("\nfor loop \n")
for i :=0; i < 6; i++ {
	fmt.Println("Current value in for loop is ", i)
}

fmt.Println("\nforEach loop \n")
arrayEx := []int64{ 1907,2000,1883}
for place,value := range arrayEx {
	fmt.Println("Place : ", place , " Value : ", value)
}

// 2. Data structures suitable for iteration
// Built in data structures are in go array, slice
fmt.Println("\n2. Data structures suitable for iteration\n")
array1 := [4]int64{24,02,2000,1907}
fmt.Println("Array is ", array1)
for i := 0; i < 4; i++ {
	fmt.Println(i+1,". element is : ", array1[i])
}

fmt.Println("\n")
slice1 := [4]string{"KING","OF","THE","NOTRH"}
for place,value := range slice1 {
	fmt.Println("Place : ", place , " Value : ", value)
}

// 3. The way the next item is accessed
fmt.Println("\n3. The way the next item is accessed\n")
num2:= 0 
    for num2 < 8 {
        if  num2 == 6 {
            fmt.Println("Reached  '6'") 
            num2 = num2 + 2
			break
        }
        num2= num2 + 1
        fmt.Println(" Turn ", num2 , " times") 
    }
    fmt.Println("\n")


    for k := 0; k < 11; k = k +2 {
        if  k == 6 {
            fmt.Println("Reached  '6'") 
			break
        }
        fmt.Println(" Turn ", k , " times") 
    }

}
